packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -264,13 +274,77 @@ export function canvasReducer(
         };
       });
     }
+    case 'moveLayersIntoParentAtPoint': {
+      const [, , point] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const indexPaths = getSelectedLayerIndexPathsExcludingDescendants(state);
+
+      const parentIndexPath = Layers.findIndexPath(
+        state.sketch.pages[pageIndex],
+        (layer) =>
+          (Layers.isArtboard(layer) || Layers.isSymbolMaster(layer)) &&
+          rectContainsPoint(layer.frame, point),
+      );
+
+      return produce(state, (draft) => {
+        const draftPage = draft.sketch.pages[pageIndex];
+
+        const layerInfo = indexPaths.map((indexPath) => ({
+          layer: Layers.access(draftPage, indexPath) as Layers.ChildLayer,
+          parent: getParentLayer(draftPage, indexPath),
+          transform: getLayerTransformAtIndexPath(draftPage, indexPath),
+        }));
+
+        const parent = parentIndexPath
+          ? (Layers.access(draftPage, parentIndexPath) as Layers.ParentLayer)
+          : draftPage;
+
+        if (
+          layerInfo.every((l) => l.parent === parent) ||
+          layerInfo.some(
+            (l) => Layers.isArtboard(l.layer) || Layers.isSymbolMaster(l.layer),
+          )
+        )
+          return;
+
+        deleteLayers(indexPaths, draftPage);
```

Comment:
If we delete layers, we won't be able to use the `parentIndexPath` anymore, since that will potentially change the layer hierarchy.